environment/src/test/java/com/sequenceiq/environment/credential/service/CredentialServiceTest.java

```diff
@@ -0,0 +1,224 @@
+package com.sequenceiq.environment.credential.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyCollection;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.ws.rs.BadRequestException;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.cloudbreak.common.json.JsonUtil;
+import com.sequenceiq.cloudbreak.message.CloudbreakMessagesService;
+import com.sequenceiq.cloudbreak.service.secret.service.SecretService;
+import com.sequenceiq.environment.credential.attributes.CredentialAttributes;
+import com.sequenceiq.environment.credential.attributes.azure.AzureCredentialAttributes;
+import com.sequenceiq.environment.credential.attributes.azure.CodeGrantFlowAttributes;
+import com.sequenceiq.environment.credential.domain.Credential;
+import com.sequenceiq.environment.credential.repository.CredentialRepository;
+import com.sequenceiq.environment.credential.validation.CredentialValidator;
+import com.sequenceiq.notification.NotificationSender;
+
+@ExtendWith(SpringExtension.class)
+public class CredentialServiceTest {
+    private static final Credential CREDENTIAL;
+
+    private static final String PLATFORM = "PLATFORM";
+
+    @Inject
+    private CredentialService credentialServiceUnderTest;
+
+    @MockBean
+    private CredentialRepository repository;
+
+    @MockBean
+    private CredentialValidator credentialValidator;
+
+    @MockBean
+    private ServiceProviderCredentialAdapter credentialAdapter;
+
+    @MockBean
+    private CredentialPrerequisiteService credentialPrerequisiteService;
+
+    @MockBean
+    private SecretService secretService;
+
+    @MockBean
+    private NotificationSender notificationSender;
+
+    @MockBean
+    private CloudbreakMessagesService cloudbreakMessagesService;
+
+    static {
+        CREDENTIAL = new Credential();
```

Comment:
What is the reason for static init block?

Reply:
no real reason, I created test data as static field and some of them needs initialization

Reply:
no, it is a mistake, I'll fix it